src/Detector/Node/NodeDetector.cs

```diff
@@ -122,9 +135,26 @@ public PlatformDetectorResult Detect(DetectorContext context)
                 HasLernaJsonFile = hasLernaJsonFile,
                 HasLageConfigJSFile = hasLageConfigJSFile,
                 LernaNpmClient = lernaNpmClient,
+                HasYarnrcYmlFile = hasYarnrcYmlFile,
+                IsYarnLockFileValidYamlFormat = IsYarnLockFileValidYamlFormat,
             };
         }
 
+        private bool IsYarnLockFileYamlFile(ISourceRepo sourceRepo, string filePath)
+        {
+            var yamlContent = sourceRepo.ReadFile(filePath);
+            var yamlStream = new YamlStream();
+            try
+            {
+                yamlStream.Load(new StringReader(yamlContent));
```

Comment:
should we do any check to make sure it is indeed a yarn 2 file?

Reply:
It's used to parse yarn.lock file, if it's a valid YAML then it'll not throw any exceptions. I added tests for this part to make sure it works good.